djvm/src/main/java/sandbox/java/util/Currency.java

```diff
@@ -0,0 +1,89 @@
+package sandbox.java.util;
+
+import sandbox.java.io.BufferedInputStream;
+import sandbox.java.io.DataInputStream;
+import sandbox.java.io.InputStream;
+import sandbox.java.lang.DJVM;
+import sandbox.java.lang.String;
+import sandbox.java.security.AccessController;
+import sandbox.java.security.PrivilegedAction;
+
+import java.io.IOException;
+import java.io.Serializable;
+
+/**
+ * This is a dummy class that implements just enough of {@link java.util.Currency}
+ * to allow us to compile {@link sandbox.java.util.Currency.1}.
+ */
+@SuppressWarnings({"unused", "RedundantThrows", "WeakerAccess"})
+public final class Currency extends sandbox.java.lang.Object implements Serializable {
+    private static final java.lang.String NOT_IMPLEMENTED = "Dummy class - not implemented";
+
+    private static final int MAGIC_NUMBER = 0x43757244;
+    private static final int A_TO_Z = ('Z' - 'A') + 1;
+    private static final int VALID_FORMAT_VERSION = 2;
+
+    static int formatVersion;
+    static int dataVersion;
+    static int[] mainTable;
+    static long[] scCutOverTimes;
+    static String[] scOldCurrencies;
+    static String[] scNewCurrencies;
+    static int[] scOldCurrenciesDFD;
+    static int[] scNewCurrenciesDFD;
+    static int[] scOldCurrenciesNumericCode;
+    static int[] scNewCurrenciesNumericCode;
+    static String otherCurrencies;
+    static int[] otherCurrenciesDFD;
+    static int[] otherCurrenciesNumericCode;
+
+    static {
+        AccessController.doPrivileged(new PrivilegedAction<Void>() {
+            @Override
+            public Void run() {
```

Comment:
Same point as in previous PR, a _dummy_ class is equivalent to a compiler stub by definition. If it has any logic into it then it is *not* a dummy class. It is not pedantry: how can i guess what is the logic in this constructor relevant for?

Reply:
`java.util.Currency` is a dummy class, but `java.util.Currency$1` is _not_:
```java
// Compiles to java.util.Currency$1
new PrivilegedAction<Void>() {
    @Override
    public void run() {
        // etc etc
    }
}
```